hs_file_types/models/geofeature.py

```diff
@@ -155,6 +155,24 @@ def get_preview_data_url(self, resource, folder_path):
 
         return preview_data_url
 
+    def check_valid_geoserver_projection(self):
+        """Check if the content is using a projection supported by GeoServer."""
+
+        try:
+            proj = osr.SpatialReference(wkt=str(self.originalcoverage.projection_string))
+            authority = ':'.join((proj.GetAttrValue('AUTHORITY', 0), proj.GetAttrValue('AUTHORITY', 1),))
+
+            with open(os.path.dirname(__file__) + '/../geoserver_epsg.txt') as f:
```

Comment:
Where would you recommend storing that type of metadata? I asked Alva about that when we first set up GeoServer and he said HydroShare didn't have a way to handle system metadata that isn't modifiable by users. That's why we went the route of generating the GeoServer URLs each time the page was loaded instead of doing it once and storing them somewhere.

Reply:
There is an extra_data field on each logical file for this type of thing.  @hyi - used it for storing the URL in a .url file.

https://github.com/hydroshare/hydroshare/blob/develop/hs_file_types/models/base.py#L723

Reply:
I've moved the projection checks to the update_web_services task, so the check will only happen when web services are updated, and the result is stored in extra_data: https://github.com/hydroshare/hydroshare/pull/4311/commits/0c1176adc9001ccf39b2a2958b7b01e320446c6a